src/vm.rs

```diff
@@ -518,27 +518,20 @@ impl<'a, V: Verifier, E: UserDefinedError, I: InstructionMeter> EbpfVm<'a, V, E,
         .collect();
         let (program_vm_addr, program) = executable.get_text_bytes();
         let number_of_syscalls = executable.get_syscall_registry().get_number_of_syscalls();
-        let mut vm = EbpfVm {
+        let vm = EbpfVm {
             verified_executable,
             program,
             program_vm_addr,
-            memory_mapping: MemoryMapping::new(regions, config)?,
-            tracer: Tracer::default(),
-            syscall_context_objects: vec![
-                std::ptr::null_mut();
-                SYSCALL_CONTEXT_OBJECTS_OFFSET + number_of_syscalls
-            ],
+            program_environment: ProgramEnvironment {
+                memory_mapping: MemoryMapping::new(regions, config)?,
+                syscalls: [ptr::null_mut(); SyscallRegistry::MAX_SYSCALLS],
```

Comment:
I think this should still be referred to as `syscall_context_objects` because `syscall` usually means the function pointers.

Reply:
Ah yeah, fair enough